ledger/test-common/src/main/daml/benchtool/InterfaceSubscription.daml

```diff
@@ -0,0 +1,60 @@
+-- Copyright (c) 2022 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+-- SPDX-License-Identifier: Apache-2.0
+
+module InterfaceSubscription where
+
+import Foo
+
+data FooData = FooData
+  with
+    signatory : Party
+    observers : [Party]
+    payload : Text
+    keyId: Text
+    templateName: Text
+  deriving (Eq, Show)
+
+interface FooI1 where
+  viewtype FooData
+
+  interface instance FooI1 for Foo.Foo1 where
+    view = FooData with templateName = "Foo1", ..
+
+
+foo2ToFooData : Foo.Foo2 -> FooData
+foo2ToFooData Foo.Foo2{..} = FooData with templateName = "Foo2", ..
+
+fooDataToFoo2 : FooData -> Foo.Foo2
+fooDataToFoo2 FooData{..}
+  | templateName == "Foo2" = Foo.Foo2 {..}
+  | otherwise = error "fooDataToFoo2 called non non-foo2"
+
+foo2Roundtrip : Int -> Foo.Foo2 -> Foo.Foo2
```

Comment:
Is this needed?
I think it's best these Benchtool oriented daml files a lean as possible

Reply:
Could you also add some comments what you're trying to achieve in respective code sections?

Reply:
Yes it is needed to be able to benchmark how views of different complexity are being computed.

I have added a comment to explain every case (FooI1, FooI2, FooI3)